numpy/testing/utils.py

```diff
@@ -726,14 +725,13 @@ def chk_same_position(x_id, y_id, hasval='nan'):
                 raise AssertionError(msg)
 
         if isnumber(x) and isnumber(y):
-            x_id, y_id = zeros_like(x, dtype=bool_), zeros_like(y, dtype=bool_)
```

Comment:
A more trivial change would replace just this line with
```
x_id = zeros(x.shape, dtype=bool_)
y_id = zeros(y.shape, dtype=bool_)
```
but it seems unnecessary to create these arrays in the first place -- and indeed the code becomes quite a bit shorter and, to me at least, clearer by just removing elements from the arrays.